src/lib/selectors.ts

```diff
@@ -0,0 +1,622 @@
+      }
+      channel.freeze()
+        .then(() => {
+          resolve();
+          // Add pubSub process
+          /**
+           * consider divide the logic to
+           * _freezeGroupChannel and _freezeOpenChannel
+           */
+        })
+        .catch(reject)
+    })
+  )
+)
+
+/**
+ * const unfreezeChannel = selectors.unfreezeChannel(currentChannel);
+ * unfreezeChannel()
+ *  .then((channel) => {})
+ *  .catch((error) => {})
+ */
+export const unfreezeChannel = (channel: GroupChannel | OpenChannel) => (
+  (): Promise<void> => (
+    new Promise((resolve, reject) => {
+      if (!(channel.unfreeze && typeof channel?.unfreeze === 'function')) {
+        reject(new Error('Not found the function "unfreeze"'));
+      }
+      channel.unfreeze()
+        .then(() => {
+          resolve();
+          // Add pubSub process
+          /**
+           * consider divide the logic to
+           * _unfreezeGroupChannel and _unfreezeOpenChannel
+           */
+        })
+        .catch(reject)
+    })
+  )
+)
+
+// 4. Message
+export class UikitMessageHandler {
+  private _onPending: MessageHandler = noop;
+  private _onFailed: FailedMessageHandler = noop;
+  private _onSucceeded: MessageHandler = noop;
+
+  public triggerPending(message: SendableMessage): void {
+    this._onPending(message);
+  }
+  public triggerFailed(error: Error, message: SendableMessage): void {
+    this._onFailed(error, message.isResendable ? message : null);
+  }
+  public triggerSucceeded(message: SendableMessage): void {
+    this._onSucceeded(message);
+  }
+
+  public onPending(handler: MessageHandler): UikitMessageHandler {
+    if (typeof handler === 'function') {
+      this._onPending = handler;
+    }
+    return this;
+  }
+  public onFailed(handler: FailedMessageHandler): UikitMessageHandler {
+    if (typeof handler === 'function') {
+      this._onFailed = handler;
+    }
+    return this;
+  }
+  public onSucceeded(handler: MessageHandler): UikitMessageHandler {
+    if (typeof handler === 'function') {
+      this._onSucceeded = handler;
+    }
+    return this;
+  }
+}
+
+/**
+ * const sendUserMessage = selectors.sendUserMessage(state);
+ * sendUserMessage(
+ *  channel: GroupChannel | OpenChannel,
+ *  params: UserMessageCreateParams,
+ * )
+ *  .onPending((message) => {})
+ *  .onFailed((error, message) => {})
+ *  .onSucceeded((message) => {})
+ */
+
+export const sendUserMessage = (state: SendBirdState) => (
+  (channel: GroupChannel | OpenChannel, params: UserMessageCreateParams): UikitMessageHandler => {
+    const handler = new UikitMessageHandler();
+    const pubSub = getPubSub(state);
+    channel.sendUserMessage(params)
+      .onFailed((error, message) => handler.triggerFailed(error, message))
+      .onPending((message) => {
+        pubSub.publish(
+          topics.SEND_MESSAGE_START,
+          { message, channel },
+        );
+        handler.triggerPending(message);
+      })
+      .onSucceeded((message) => {
```

Comment:
onSuccess is probably a better naming, but SDK already used onSucceeded.. sooo not  a good idea to move